### PR TITLE
Adding promise

### PR DESCRIPTION
**Pull request by tracetoio-weilong**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,72 +1,61 @@
...
   constructor(ipfs_url = {host: 'ipfs.alpha.traceto.io', port: '443', protocol: 'https'}){
-    this.ipfs = ipfsAPI(ipfs_url);
+    this.ipfs = ipfsClient(ipfs_url);
   }
 
   GeneratePassphrase(len){
     this.passPhrase = secrets.random(len);
   }
 
   setPassphraseForWeb(seed, callback){
```
Comment (tracetoio-dias) [inline index.js]: Do we need callback here?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "tracetoio-crypto-lib",
-  "version": "1.1.1",
+  "version": "1.2.2",
```
Comment (tracetoio-dias) [inline package.json]: This seems to break if the parameters are different?

File: package.json [hunk 3]
```diff
@@ -18,7 +18,7 @@
     "chai": "^4.1.2",
     "eth-ecies": "^1.0.3",
     "ethereumjs-util": "^5.2.0",
-    "ipfs-api": "^24.0.1",
+    "ipfs-http-client": "^29.1.1",
```
Comment (tracetoio-dias) [inline package.json:21]: I think if you  are changing this , it is good to add a changelog or add some info to readme.md

File: index.js (hunk 1 above)
Comment (tracetoio-weilong) [reply index.js]: removed

File: package.json (hunk 2 above)
Comment (tracetoio-weilong) [reply package.json]: updated

**Review by tracetoio-dias — APPROVED** [review]

**Pull request merged**